evm_loader/lib/src/abi/mod.rs

```diff
@@ -0,0 +1,186 @@
+mod cancel_trx;
+mod collect_treasury;
+mod context;
+mod create_ether_account;
+mod deposit;
+mod emulate;
+mod get_ether_account_data;
+mod get_neon_elf;
+mod get_storage_at;
+mod init_environment;
+mod trace;
+
+use self::context::AbiContext;
+use crate::{
+    commands::get_neon_elf::CachedElfParams,
+    config::{self},
+    types::request_models::{EmulationParamsRequestModel, RequestWithSlot},
+    Config, Context, LibMethods, NeonError,
+};
+use abi_stable::{
+    prefix_type::WithMetadata,
+    sabi_extern_fn,
+    std_types::{RStr, RString},
+};
+use async_ffi::FutureExt;
+use evm_loader::types::Address;
+use lazy_static::lazy_static;
+use neon_lib_interface::{
+    types::{NeonEVMLibError, RNeonEVMLibResult},
+    NeonEVMLib,
+};
+use solana_sdk::pubkey::Pubkey;
+use std::str::FromStr;
+
+lazy_static! {
+    static ref RT: tokio::runtime::Runtime = tokio::runtime::Runtime::new().unwrap();
+}
+
+pub const _MODULE_WM_: &WithMetadata<NeonEVMLib> = &WithMetadata::new(NeonEVMLib {
+    hash,
+    get_version,
+    invoke,
+});
+
+#[sabi_extern_fn]
+fn hash() -> RString {
+    env!("NEON_REVISION").into()
+}
+
+#[sabi_extern_fn]
+fn get_version() -> RString {
+    env!("CARGO_PKG_VERSION").into()
+}
+
+#[sabi_extern_fn]
+fn invoke<'a>(method: RStr<'a>, params: RStr<'a>) -> RNeonEVMLibResult<'a> {
+    async move {
+        RT.block_on(dispatch(method.as_str(), params.as_str()))
+            .map(RString::from)
+            .map_err(neon_error_to_rstring)
+            .into()
+    }
+    .into_local_ffi()
+}
```

Comment:
This introduces the [`async-sync-async` sandwich problem](https://rust-lang.github.io/wg-async/vision/submitted_stories/shiny_future/barbara_enjoys_an_async_sandwich.html) that I removed in #158, because:
- `neon-rpc` async context calls `invoke`- async;
- `invoke` blocks the current thread with `RT.block_on` - sync;
- `RT.block_on` in turn calls into another async context (`dispatch(method.as_str(), params.as_str())` call) - async.

There is no need to call `RT.block_on()` on the future returned by `dispatch(method.as_str(), params.as_str())`. It can simply be `.await`ed. The `LocalBorrowingFfiFuture` returned by `invoke` will simply be `poll`ed on the `tokio` runtime of `neon-rpc` server.

The `tokio = { version = "1", features = ["full"] }` dependency can be removed from `neon-lib` crate, since it is never used anymore. So the code of `neon-lib` crate is async-runtime agnostic, as it should be.

Reply:
tokio is used in lib/src/commands/get_config.rs:5:5
Thank you for block on comment.